src/twisted/newsfragments/10231.bugfix

```diff
@@ -0,0 +1 @@
+twisted.internet.defer.inlineCallbacks has an improved type annotation, to avoidtyping errors when it is used on a function which returns a non-None result.
```

Comment:
```suggestion
twisted.internet.defer.inlineCallbacks has an improved type annotation, to avoid typing errors when it is used on a function which returns a non-None result.
```